framework/include/base/MooseError.h

```diff
@@ -18,16 +18,21 @@
 #include "Moose.h"
 #include "MooseException.h"
 
-// temporary fix to allow merging moose PR #4278 until libmesh PR #415 is merged
-#ifndef __LIBMESH_TIME__
-#  define __LIBMESH_TIME__ __TIME__
-#endif
-#ifndef __LIBMESH_DATE__
-#  define __LIBMESH_DATE__ __DATE__
-#endif
-
 // libMesh includes
 #include "libmesh/print_trace.h"
+#include "libmesh/libmesh_common.h"
+
+// C++ includes
+#include <cstdlib>
+
+/**
+ * Application abort macro. Uses MPI_Abort if available, std::abort otherwise
+ */
+#if defined(LIBMESH_HAVE_MPI)
+#define MOOSE_ABORT MPI_Abort(libMesh::GLOBAL_COMM_WORLD, 1); std::abort(); while (0)
```

Comment:
You have the `while (0)` part but not the do?  I wasn't aware this was valid C++...